### ISSUE TITLE
Sprinklers still on, but switch showing off

### ISSUE THREAD

**Issue by qwserty** [sebr/bhyve-home-assistant#26]
It isn't every time, but about 75% of the time the history shows that automation turned on the sprinklers and then a few seconds later will show the sprinklers off again. The sprinklers DO NOT shutoff, but home assistant is showing that they are off. Makes it difficult to know if everything is working like it should and that your sprinklers ran the proper length of time by checking the history.

The only thing I can see that might be causing the problem is that for some reason I'm getting two calls to turn on a station.
For example. My north side sprinkler was on when I woke up this morning and timer was showing 5 minutes left, but history shows it was turned off 2 seconds after it came on. In the log it is showing the initial call to turn on at 06:00:05 and then another call at 06:00:07 and even though it looks like they were both a call to turn on station 4 the second call put station 4's switch into an off position, but it didn't shut off the sprinklers. Hmmm.. Maybe because the switch is setup as a toggle or something???

Here is the log. No error's or anything just two calls. When the history shows correct I don't have both these calls.
```
2020-06-28 06:00:00 INFO (MainThread) [homeassistant.components.automation] Executing Irrigation Northside Yard
2020-06-28 06:00:00 INFO (MainThread) [homeassistant.components.automation] Irrigation Northside Yard: Running script
2020-06-28 06:00:00 INFO (MainThread) [homeassistant.components.automation] Irrigation Northside Yard: Executing step call service
2020-06-28 06:00:00 INFO (MainThread) [custom_components.bhyve.switch] start_watering service called
2020-06-28 06:00:00 INFO (MainThread) [custom_components.bhyve.switch] Starting watering
2020-06-28 06:00:00 INFO (MainThread) [homeassistant.components.automation] Irrigation Northside Yard: Executing step call service
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer rain delay - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Frontyard Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: SouthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Backyard Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Drip system zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Sprinkler Security zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer battery level - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer state - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Frontyard Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: SouthSide Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Sprinkler Security zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Drip system zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00
2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: Backyard Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00

2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer rain delay - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Frontyard Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: SouthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Backyard Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Drip system zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Sprinkler Security zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer battery level - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Provo House Timer state - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Frontyard Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: SouthSide Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Sprinkler Security zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Drip system zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: Backyard Sprinklers zone history - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'
```

Thank you.

**Comment by sebr** [comment]
@qwserty let me take a look - it's hard for me as I only have 1 device and 1 zone - so I have to reverse engineer support for multiple zones.

> about 75% of the time the history shows that automation turned on the sprinklers and then a few seconds later will show the sprinklers off again

Are you referring to the zone switch or the zone history entities?

> 2020-06-28 06:00:00 INFO (MainThread) [homeassistant.components.automation] Executing Irrigation Northside Yard

Can you please share the configuration for this automation?

If you could reply by adding a comment on github and _not_ an email reply that would be very helpful - email formatting is impossible to read/edit in gh.

**Comment by qwserty** [comment]
Both.. The Zone switch is off and the History is showing off (for the zone
switch), but the sprinklers will still be running until the amount of
minutes I gave it completes and if I check on the bhyve site it shows it
running with the correct amount of minutes left. Almost like that second
call to the unit which is exactly like the first call just toggles the
switch back to off, but doesn't actually shut anything down. Luckily
otherwise my lawn would only get 2 seconds of water half the time. :-)

**Comment by sebr** [comment]
According to your logs...

The zone is switched on here:

> `2020-06-28 06:00:05 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'watering_in_progress_notification', 'program': 'manual', 'current_station': 4, 'run_time': 20, 'started_watering_station_at': '2020-06-28T12:00:01.00`

Then 2 seconds later is switched off:

> `2020-06-28 06:00:07 INFO (MainThread) [custom_components.bhyve] Callback update: NorthSide Sprinklers zone - 5d3786064f0c47ece9e4fa1f - {'event': 'change_mode', 'mode': 'manual', 'program': None, 'stations': [{'station': 4, 'run_time': 20.0}], 'device_id': '5d3786064f0c47ece9e4fa1f', 'timestamp'`

So I made some errors assuming I understood what the `change_mode` event is used for :)

Can you share how you are starting the zone watering so I can try replicate?

**Comment by qwserty** [comment]
Sure.. I don't know if you want the entire setup which is very large right
now, but this is the automation.yaml for just one of the zones.. Basically
the trigger is every minute with the condition of a time & day of the week
that is set in a card on the frontend. Oh and the run time is set off the
front end as well.

You know oddly enough once I completed building the entire script to get
this all going. I have not seen the problem with history for the last few
days. I don't know why. The only difference is that I'm getting the time,
day of week and timer off variables now instead of it being hard coded. I
did have a brief problem with it shutting off as soon as it was turned on,
but found out that it was being caused by my automation script I had that
would turn on my flowerbed water for 30 seconds and then shut it off if
motion was detected to deter my dogs from ripping up our garden. Not
exactly sure why it was shutting down everything, but I guess two zone
can't be working at the same time or something. (Which is fine I can only
run one zone at a time)

#####################
# Frontyard water grass #
#####################
- id: 'Irrigation Front Yard yard 1st run'
  alias: 'Irrigation Front Yard yard 1st run'
  description: Start the Front Yard yard sprinklers
  trigger:
  - platform: time_pattern
    minutes: "/1"
  condition:
    condition: and
    conditions:
    - condition: template
      value_template: "{{ states('sensor.time') ==
          (state_attr('input_datetime.frontyard_starttime', 'timestamp') |
int |
          timestamp_custom('%H:%M', False)) }}"
    - condition: or
      conditions:
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Mon") and
(states.input_boolean.frontyard_monday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Tue") and
(states.input_boolean.frontyard_tuesday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Wed") and
(states.input_boolean.frontyard_wednesday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Thu") and
(states.input_boolean.frontyard_thursday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Fri") and
(states.input_boolean.frontyard_friday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Sat") and
(states.input_boolean.frontyard_saturday.state=="on")}}'
      - condition: template
        value_template: '{{ (now().strftime("%a") == "Sun") and
(states.input_boolean.frontyard_sunday.state=="on")}}'
  action:
  - service: bhyve.start_watering
    data_template:
      entity_id: switch.frontyard_sprinklers_zone_2
      minutes: "{{ states('input_number.frontyard_runtime') | int }}"
  - service: timer.start
    data_template:
      entity_id: timer.backyard_zone_finish
      duration: "00:{{ states.input_number.frontyard_runtime.state | int
}}:00"

**Comment by sebr** [comment]
I've released a fix in version 2.0.2, thanks for your patience!
